lib/file.c

```diff
@@ -393,6 +393,8 @@ static CURLcode file_do(struct Curl_easy *data, bool *done)
   if(-1 != fstat(fd, &statbuf)) {
     if(!S_ISDIR(statbuf.st_mode))
       expected_size = statbuf.st_size;
+    else
+      expected_size = 0;
```

Comment:
nit: Why not initialize at variable declaration?

Reply:
I was erring towards leaving the existing behaviour as-is; the variable is already initialized to -1 at variable declaration.

Arguably, it makes _more_ sense to leave it at -1 - but expected_size is assumed to have a reasonable value if fstated==TRUE, which causes assertions to fail.

...But you're right, I'll rework this to avoid that assumption instead.